### PR TITLE
[RFC] Remove Pipeline::SetCompletionCallback() API

### PR DESCRIPTION
**Pull request by prak-nv**
This API is most likely unused outside of test suite.
It was never public.

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/pipeline/executor/executor_test.cc [hunk 1]
```diff
@@ -576,30 +506,24 @@ TYPED_TEST(ExecutorSyncTest, TestPrefetchedExecution) {
...
   exe->RunGPU();
 
   DeviceWorkspace ws;
-  exe->Outputs(&ws);
+  exe->ShareOutputs(&ws);
   ASSERT_EQ(ws.NumOutput(), 1);
   ASSERT_EQ(ws.NumInput(), 0);
   ASSERT_TRUE(ws.OutputIsType<GPUBackend>(0));
   test::CheckResults(ws, batch_size, 0, tl);
+  exe->ReleaseOutputs();
```
Comment (mzient) [inline dali/pipeline/executor/executor_test.cc:520]: Why this change?

Comment (prak-nv) [reply dali/pipeline/executor/executor_test.cc:520]: Wanted to explicitly at this point ReleaseOutputs() before next run.
From functionality perspective doesn't change the semantics, since next exe->Outputs() would call that for but after next run.

Comment (prak-nv) [reply dali/pipeline/executor/executor_test.cc:520]: NVM can probably revert that change. It is pointless here.

Comment (mzient) [reply dali/pipeline/executor/executor_test.cc:520]: Actually, a call to Otuputs calls Release first, then Share. But whatever - if it works...

**Review by mzient — APPROVED** [review]

**Comment by prak-nv** [comment]
!build
